usaspending_api/broker/management/commands/load_fpds_transactions.py

```diff
@@ -0,0 +1,172 @@
+from django.core.management.base import BaseCommand
+import logging
+import re
+import psycopg2
+from datetime import datetime, timezone
+
+from usaspending_api.etl.transaction_loaders.fpds_loader import load_ids, destroy_orphans, delete_stale_fpds
+from usaspending_api.common.retrieve_file_from_uri import RetrieveFileFromUri
+from usaspending_api.common.helpers.date_helper import datetime_command_line_argument_type
+from usaspending_api.common.helpers.sql_helpers import get_broker_dsn_string
+from usaspending_api.common.helpers.etl_helpers import update_c_to_d_linkages
+from usaspending_api.etl.award_helpers import update_awards, update_contract_awards
+from usaspending_api.broker.helpers.last_load_date import get_last_load_date, update_last_load_date
+
+logger = logging.getLogger("console")
+
+BROKER_CONNECTION_STRING = get_broker_dsn_string()
+
+CHUNK_SIZE = 5000
+
+ALL_FPDS_QUERY = "SELECT {} FROM detached_award_procurement"
+
+
+class Command(BaseCommand):
+    help = "Sync USAspending DB FPDS data using Broker for new or modified records and S3 for deleted IDs"
+
+    modified_award_ids = []
+
+    @staticmethod
+    def get_cursor_for_date_query(connection, date, count=False):
+        if count:
+            db_cursor = connection.cursor()
+            db_query = ALL_FPDS_QUERY.format("COUNT(*)")
+        else:
+            db_cursor = connection.cursor("fpds_load", cursor_factory=psycopg2.extras.DictCursor)
+            db_query = ALL_FPDS_QUERY.format("detached_award_procurement_id")
+
+        if date:
+            db_cursor.execute(db_query + " WHERE updated_at >= %s", [date])
+        else:
+            db_cursor.execute(db_query)
+        return db_cursor
+
+    def load_fpds_incrementally(self, date):
+        """
+        Handles "increment" loads, which includes deleting transactions so long as a date is provided
+        :param date: if provided, loader will load and delete all transactions starting from this date
+        :return: nothing
+        """
+        if date is None:
+            logger.info("fetching all fpds transactions...")
+        else:
+            logger.info("fetching fpds transactions since {}...".format(str(date)))
+
+        # First clear any transactions marked for deletion. If the transactions have since been re-added, they will be back in the broker DB and will be re-inserted in the next step
+        stale_awards = delete_stale_fpds(date.date())
```

Comment:
Hit an issue here because it is passing `data.date()` but `date` has the potential to be None. When `date` is None it errors out.